sidechain/consensus/aura/src/peer_block_sync.rs

```diff
@@ -0,0 +1,214 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::block_production_suspension::{IsBlockProductionSuspended, SuspendBlockProduction};
+use core::marker::PhantomData;
+use its_consensus_common::{BlockImport, Error, Result};
+use its_primitives::{traits::SignedBlock as SignedSidechainBlockTrait, types::BlockHash};
+use log::*;
+use sp_runtime::traits::{Block as ParentchainBlockTrait, Header as ParentchainHeaderTrait};
+use std::{sync::Arc, vec::Vec};
+
+/// Trait for syncing sidechain blocks from a peer validateer.
+///
+/// This entails importing blocks and detecting if we're out of date with our blocks, in which
+/// case we fetch the missing blocks from a peer.
+pub trait SyncBlockFromPeer<ParentchainHeader, SignedSidechainBlock>
+where
+	ParentchainHeader: ParentchainHeaderTrait,
+	SignedSidechainBlock: SignedSidechainBlockTrait,
+{
+	fn attempt_block_sync(
```

Comment:
Minor naming detail: rust's convention is to use `try` to distinguish fallible functions from infallible ones. I suggest using `try` here too. I am also ok to just call it `block_sync`, as we don't have an infallible counterpart.

Reply:
It's not an operation in the `try` sense that will execute an operation and potentially fail. We try to import a block, and if we encounter an ancestry mismatch error, we sync blocks from a peer. But I'm also not super happy with the naming, I can name it `sync_blocks` for now to keep it simple.